index/scorch/snapshot_index.go

```diff
@@ -507,11 +523,29 @@ func (i *IndexSnapshot) documentVisitFieldTerms(id index.IndexInternalID,
 	ss := i.segment[segmentIndex]
 
 	if zaps, ok := ss.segment.(segment.DocumentFieldTermVisitable); ok {
-		// get the list of doc value persisted fields
-		pFields, err := zaps.VisitableDocValueFields()
-		if err != nil {
-			return nil, err
+		pFields := ss.cachedDocs.getPersistedFields()
+		// if all fields are dv presisted
+		if len(pFields) > 0 && checkSubset(pFields, fields) {
```

Comment:
The multiple special case checking here might be repetitious...

It has a bit of feel of a math library, where it logically handles special cases... like, "first, let me handle all the even integers", "next, we're going to handle any odd integers", "finally, let me handle 0 as the final case" -- such an approach is logically correct, but doesn't have a minimal, concise feel?

Reply:
I agree to your point, and that will definitely be elegant.
 But let me think of another perspective here.. these simpler procedure kind of logic has less cognitive load at a later point while debugging. You need to load less into your mind..
Else, lets say we need to jump into all those visitCache, zaps Visits function codes to figure out whether things work correctly if we pass the whole set of fields..where is it getting filled..or from where fields are getting served etc... need to load all that flow while debugging at a later point.. :)